src/datasets/arrow_dataset.py

```diff
@@ -440,25 +476,24 @@ def split_features_and_labels(input_batch):
 
             tf_dataset = tf_dataset.map(split_features_and_labels)
 
-        elif len(columns) == 1:
+        elif isinstance(tf_dataset.element_spec, dict) and len(tf_dataset.element_spec) == 1:
```

Comment:
(nit)

Not a big fan of changing the return type of a function based on the content of a dataset. I'd rather make the output type deterministic based on the input parameters only.

What if we do that only if the type of the `columns` argument of `to_tf_dataset` is a string, and if the user passes a list with a single element, then it should stay a dict with a single element ?

Reply:
I think that's a good idea - it would definitely have been better to do it that way, but it might break backward compatibility for some of our existing examples if we change now. Let me do some testing and see how bad it'd be!

Reply:
Update: There are definitely some issues with this change, but my updates to train_step and test_step in another PR fix a lot of it. Can we leave it for now (since this method already exists, it's not like this PR makes it worse!), and maybe deprecate or change the behaviour in a follow-up PR?

Reply:
Yup sounds good :)

Reply:
Sure! I'll work on adding the tests you suggested and then ping you for another review.